lib/vagrant-parallels/driver/base.rb

```diff
@@ -432,12 +433,13 @@ def read_guest_ip_dhcp
         end
 
         # Returns an IP of the virtual machine fetched from prlctl.
+        # Returns an empty string if the IP coudn't be determined this way.
         #
         # @return [String] IP address returned by `prlctl list -f` command
         def read_guest_ip_prlctl
           vm_info = json { execute_prlctl('list', @uuid, '--full', '--json') }
-          ip = vm_info.first.fetch('ip_configured', nil)
-          ip == '-' ? nil : ip
+          ip = vm_info.first.fetch('ip_configured', '')
+          ip == '-' ? '' : ip
```

Comment:
I think nil is apt for an easy return from the ssh_ip function. If we change to nil in dhcp_ function also, we can return by just prlctl_ || dhcp_

Reply:
I prefer to keep `read_guest_ip_dhcp` and `read_guest_ip_prlctl` behaving in the same way. The existing `read_guest_ip_dhcp` already returns an empty string on a failed attempt, then `read_guest_ip_prlctl` should do the same.  

If we switch one to return `nil` in that case, then we should update a second function too, to keep the code consistent. In this case I just preferred to do not touch the older function and updates the new one accordingly.

Reply:
Agreed.